application/src/main/java/org/togetherjava/tjbot/commands/mathcommands/wolframalpha/WolframAlphaCommand.java

```diff
@@ -0,0 +1,374 @@
+                        images.clear();
+                        ImageIO.write(combinedImage, "png", Path
+                            .of("C:\\Users\\Abc\\IdeaProjects\\TJ-Bot-baseRepo\\application\\src\\main\\java\\org\\togetherjava\\tjbot\\commands\\mathcommands\\wolframalpha\\sentImage%d.png"
+                                .formatted(++filesAttached))
+                            .toFile());
+                        if (message == null) {
+                            message = channel.sendFile(WolfCommandUtils.imageToBytes(combinedImage),
+                                    "result%d.png".formatted(++filesAttached));
+                        } else {
+                            message.addFile(WolfCommandUtils.imageToBytes(combinedImage),
+                                    "result%d.png".formatted(++filesAttached));
+                        }
+                        /*
+                         * messages.add(channel.sendFile(WolfCommandUtils.imageToBytes(combinedImage
+                         * ), name));
+                         */
+                        /*
+                         * bytes.add(WolfCommandUtils.imageToBytes(combinedImage));//
+                         * filesAttached++;
+                         */
+                        embeds.add(embedBuilder
+                            .setImage("attachment://result%d.png".formatted(filesAttached))
+                            .build());
+                    }
+                    resultHeight += readImage.getHeight();
+                    images.add(readImage);
+                    logger.info(
+                            "Max Width {} Result Height {} Current Width {} Current Height {} Images {}",
+                            maxWidth, resultHeight, readImage.getWidth(), readImage.getHeight(),
+                            filesAttached);
+                } catch (IOException e) {
+                    event.reply("Unable to generate message based on the WolframAlpha response")
+                        .setEphemeral(true)
+                        .queue();
+                    logger.error("Failed to read image {} from the WolframAlpha response", image,
+                            e);
+                    return Optional.empty();
+                }
+            }
+            logger.info("Exiting pod number {}", i);
+        }
+        /*
+         * for (int i = bytes.size() - 1; i >= 0; i--) { try { ImageIO.write(ImageIO.read(new
+         * ByteArrayInputStream(bytes.get(i))), "png", Path .of(
+         * "C:\\Users\\Abc\\IdeaProjects\\TJ-Bot-baseRepo\\application\\src\\main\\java\\org\\togetherjava\\tjbot\\commands\\mathcommands\\wolframalpha\\sentImage%d.png"
+         * .formatted(i)) .toFile()); } catch (IOException e) { e.printStackTrace(); } action =
+         * action.addFile(bytes.get(i), names.get(i)); }
+         */
+        messages.add(message.setEmbeds(embeds));
+        return Optional.of(messages);
```

Comment:
```suggestion
        return messages;
```
When you're wrapping a Collection in an optional something is going wrong.

If no items are found, an empty collection should be returned instead of null.

Reply:
This is a draft rn. I have so much commented code, which is obsolete but kept in case it is needed. I will address these design issues later.

Reply:
👍